packages/vscode-extension/src/devices/IosSimulatorDevice.ts

```diff
@@ -464,26 +482,29 @@ export async function listSimulators(
 
   const runtimes = await getAvailableIosRuntimes();
 
-  const simulators = devicesPerRuntime
-    .map(([runtimeID, devices]) => {
+  const simulators = await Promise.all(
+    devicesPerRuntime.map(async ([runtimeID, devices]) => {
       const runtime = runtimes.find((item) => item.identifier === runtimeID);
 
-      return devices.map((device) => {
-        return {
-          id: `ios-${device.udid}`,
-          platform: DevicePlatform.IOS as const,
-          UDID: device.udid,
-          name: device.name,
-          systemName: runtime?.name ?? "Unknown",
-          available: device.isAvailable ?? false,
-          deviceIdentifier: device.deviceTypeIdentifier,
-          runtimeInfo: runtime!,
-        };
-      });
+      const deviceInfos = await Promise.all(
+        devices.map(async (device) => {
```

Comment:
why is this async if it doesnt do anything that's asynchronous?

This way you could eliminate Promise.all above and the one from line 485